x/auth/tx/module/module.go

```diff
@@ -0,0 +1,103 @@
+package tx
+
+import (
+	"fmt"
+
+	modulev1 "cosmossdk.io/api/cosmos/tx/module/v1"
+	"cosmossdk.io/core/appmodule"
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/depinject"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/ante"
+	"github.com/cosmos/cosmos-sdk/x/auth/posthandler"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	feegrantkeeper "github.com/cosmos/cosmos-sdk/x/feegrant/keeper"
+)
+
+func init() {
+	appmodule.Register(&modulev1.Module{},
+		appmodule.Provide(provideModule),
+	)
+}
+
+type txInputs struct {
+	depinject.In
+
+	Config   *modulev1.Module
+	TxConfig *client.TxConfig
```

Comment:
`TxConfig` needs to be an output not an input of the tx module. The tx module probably needs to take a `codec.Codec` as input to build the `TxConfig`

Reply:
Alright I'll revert